Extras/InverseDynamics/CoilCreator.cpp

```diff
@@ -11,25 +11,25 @@ CoilCreator::CoilCreator(int n) : m_num_bodies(n), m_parent(n) {
     // DH parameters (that's what's in the paper ...)
     const idScalar theta_DH = 0;
     const idScalar d_DH = 0.0;
-    const idScalar a_DH = 1.0 / m_num_bodies;
-    const idScalar alpha_DH = 5.0 * BT_ID_PI / m_num_bodies;
+    const idScalar a_DH = static_cast<idScalar>(1.0 / m_num_bodies);
+    const idScalar alpha_DH = static_cast<idScalar>(5.0 * BT_ID_PI / m_num_bodies);
     getVecMatFromDH(theta_DH, d_DH, a_DH, alpha_DH, &m_parent_r_parent_body_ref,
                     &m_body_T_parent_ref);
     // always z-axis
     m_body_axis_of_motion(0) = 0.0;
     m_body_axis_of_motion(1) = 0.0;
     m_body_axis_of_motion(2) = 1.0;
 
-    m_mass = 1.0 / m_num_bodies;
-    m_body_r_body_com(0) = 1.0 / (2.0 * m_num_bodies);
+    m_mass = static_cast<idScalar>(1.0 / m_num_bodies);
+    m_body_r_body_com(0) = static_cast<idScalar>(1.0 / (2.0 * m_num_bodies));
```

Comment:
static_cast used in many place in bullet code

https://github.com/bulletphysics/bullet3/search?utf8=✓&q=static_cast&type=